src/main/java/htsjdk/samtools/reference/AbstractIndexedFastaSequenceFile.java

```diff
@@ -146,6 +147,41 @@ protected static void sanityCheckDictionaryAgainstIndex(final String fastaFile,
         }
     }
 
+    /** Do some basic checking to make sure the fasta and the index match.
+     * <p>
+     * checks that the length of the fasta file is at least as long as the index proclaims
+     * and that beyond the last position references in the index there is only one line followed by whitespaces
+     *
+     * @param fastaFile Used for error reporting only.
+     * @param index index file to check against the dictionary.
+     */
+    public static void sanityCheckFastaAgainstIndex(final String fastaFile,
+                                                    final FastaSequenceIndex FastaSequenceIndex) {
+
+
+        final Iterator<FastaSequenceIndexEntry> iterator = FastaSequenceIndex.iterator();
+        FastaSequenceIndexEntry fastaSequenceIndex = null;
+        while (iterator.hasNext()) {
+            fastaSequenceIndex = iterator.next();
+        }
+        assert fastaSequenceIndex != null;
+        final long lastSequenceLength = fastaSequenceIndex.getSize();
+        final long lastSequenceStart = fastaSequenceIndex.getLocation();
+        final long lastSequenceEnd = lastSequenceStart + fastaSequenceIndex.getOffset(lastSequenceLength);
+
+        final long fastaLength = new File(fastaFile).length();
```

Comment:
Is this safe to do?  Does `AbstractIndexedFastaSequenceFile` support or have sub-classes that support URL based access (like http:// or ftp://)?  If so ... is there a method to call to get the file/object length?